index.js

```diff
@@ -0,0 +1,148 @@
+window.onload = (event) => {
+  var game = new Game();
+  document.getElementById('score-zero').onclick = () => {
+    updateFrame(0);
+    updateScoreBox();
+    updatePinBox();
+  };
+
+  document.getElementById('score-one').onclick = () => {
+    updateFrame(1);
+    updateScoreBox();
+    updatePinBox();
+  };
+  document.getElementById('score-two').onclick = () => {
+    updateFrame(2);
+    updateScoreBox();
+    updatePinBox();
+  };
+
+  document.getElementById('score-three').onclick = () => {
+    updateFrame(3);
+    updateScoreBox();
+    updatePinBox();
+  };
+
+  document.getElementById('score-four').onclick = () => {
+    updateFrame(4);
+    updateScoreBox();
+    updatePinBox();
+  };
+
+  document.getElementById('score-five').onclick = () => {
+    updateFrame(5);
+    updateScoreBox();
+    updatePinBox();
+  };
+
+  document.getElementById('score-six').onclick = () => {
+    updateFrame(6);
+    updateScoreBox();
+    updatePinBox();
+  };
+
+  document.getElementById('score-seven').onclick = () => {
+    updateFrame(7);
+    updateScoreBox();
+    updatePinBox();
+  };
+  document.getElementById('score-eight').onclick = () => {
+    updateFrame(8);
+    updateScoreBox();
+    updatePinBox();
+  };
+  document.getElementById('score-nine').onclick = () => {
+    updateFrame(9);
+    updateScoreBox();
+    updatePinBox();
+  };
+  document.getElementById('score-ten').onclick = () => {
+    updateFrame(10);
+    updateScoreBox();
+    updatePinBox();
+  };
+  document.getElementById('reset').onclick = () => {
+    window.location.reload();
+  }
+
+  
+  var roll = 0; 
+
+  var round = 0; 
+
+  var updateFrame = (score) => {
+    if (round > 10) {
+      return;
+    }
+    if (round === 10) {
+      if (game.spare(9) || game.strike(9)) {
+        bonusRoll(score);
+        round++;
+      }
+    }
+    if (score === 10 && !game.finalRound(round)) { 
+      
+      game.frames[round] = [score];
+      game.updateScores(round);
+      round++; 
+      roll = 0; 
+      
+      return;
+    } else if (game.frames[round] === undefined) { 
+      game.frames[round] = [score]; 
+      roll++; 
+    } else {
+      if (!game.finalRound(round) && (game.frames[round][0] + score > 10)){
+        alert("Invalid pin. Please try again");
```

Comment:
you can replace  **alert("Invalid pin. Please try again");** with **alert(err.message)**;  so you don't have to copy the same message twice
try { 
  // code...
} catch (err) {
alert(err.message)
} 
https://javascript.info/try-catch#the-try-catch-syntax